paimon-core/src/main/java/org/apache/paimon/table/Table.java

```diff
@@ -41,6 +43,18 @@ public interface Table extends Serializable {
     /** Returns the row type of this table. */
     RowType rowType();
 
+    /** Partition keys of this table. */
+    List<String> partitionKeys();
+
+    /** Primary keys of this table. */
+    List<String> primaryKeys();
+
+    /** Options of this table. */
+    Map<String, String> options();
+
+    /** Optional comment of this table. */
+    Optional<String> comment();
+
```

Comment:
Is it necessary to public `fileIO()` ? I think it's useful for creating `SnapshotManager` and `SchemaManager`.